### PR TITLE
Remove unused commit metadata from the message log

### PR DESCRIPTION
**Pull request by kulakowski**
# Description of Changes

# API and ABI

 - [ ] This is a breaking change to the module ABI
 - [ ] This is a breaking change to the module API
 - [ ] This is a breaking change to the ClientAPI
 - [ ] This is a breaking change to the SDK API

*If the API is breaking, please state below what will break*

### CODE REVIEW

**Comment by kulakowski** [comment]
cc @joshua-spacetime

File: crates/core/src/db/messages/commit.rs [hunk 1]
```diff
@@ -94,16 +52,6 @@ impl Commit {
     pub fn encode(&self, bytes: &mut Vec<u8>) {
         bytes.reserve(self.encoded_len());
 
-        if self.parent_commit_hash.is_none() {
-            bytes.push(0);
```
[the remark this answers is not in the document]
Comment (kulakowski) [reply crates/core/src/db/messages/commit.rs:98]: Yeah, this was just the first bit, me confirming that indeed all this is currently unused. I'd love to use this as a way to exercising versioning this!

File: crates/core/src/db/messages/commit.rs [hunk 2]
```diff
@@ -1,13 +1,9 @@
 use super::transaction::Transaction;
-use crate::hash::Hash;
 use std::sync::Arc;
 
 // aka "Block" from blockchain, aka RecordBatch, aka TxBatch
 #[derive(Debug)]
 pub struct Commit {
-    pub parent_commit_hash: Option<Hash>,
```
Comment (kim) [inline crates/core/src/db/messages/commit.rs:8]: I understood this as an integrity protection mechanism, which is vital for replication. Can you explain what alternative you're offering?

**Review by kim — DISMISSED** [review]

File: crates/core/src/db/messages/commit.rs (hunk 2 above)
Comment (kulakowski) [reply crates/core/src/db/messages/commit.rs:8]: I want to emphasize that this is a draft PR for me to point out that some code is currently unused to a specific person.

**Pull request closed without merging**